pkg/cmd/reset.go

```diff
@@ -100,5 +105,40 @@ func runReset(opts *resetOpts) error {
 
 	s.Logger.Warnln("this command will require an explicit confirmation starting with the next minor release (v1.3)")
 
+	fmt.Println("The following nodes will be reset. The Kubernetes cluster running on those nodes will be permanently destroyed ")
+	for _, node := range s.Cluster.ControlPlane.Hosts {
+		fmt.Printf("\t- reset control plane node %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+	for _, node := range s.Cluster.StaticWorkers.Hosts {
+		fmt.Printf("\t- reset static worker nodes %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+
+	if s.DynamicClient == nil {
+		err = kubeconfig.BuildKubernetesClientset(s)
+		if err != nil {
+			s.Logger.Warnln("Failed to build Kubernetes client set.")
+		}
+	}
+
+	// Gather information about machine-controller managed nodes
+	machines := v1alpha1.MachineList{}
+	if err := s.DynamicClient.List(s.Context, &machines); err != nil {
+		s.Logger.Warnln("Failed to list Machines. Worker nodes will not be deleted. If there are worker nodes in the cluster, you might have to delete them manually.")
+	}
+
+	for _, machine := range machines.Items {
+		fmt.Printf("\t+ reset machine-controller managed machines %q\n", machine.Name)
+	}
```

Comment:
If we fail to build the Kubernetes clientset on L117 and try to list machines on L125, the List function might throw an NPE. That's not something that we want, and we can do something along this to solve it:

1. Try to build the clientset
2. `if err == nil` -> list machines
3. `if err != nil` -> show a warning saying that we failed to list Machines

```suggestion
	err = kubeconfig.BuildKubernetesClientset(s)
	if err == nil {
		// Gather information about machine-controller managed nodes
		machines := v1alpha1.MachineList{}
		if err := s.DynamicClient.List(s.Context, &machines); err != nil {
			s.Logger.Warnln("Failed to list Machines. Worker nodes will not be deleted. If there are worker nodes in the cluster, you might have to delete them manually.")
		}

		for _, machine := range machines.Items {
			fmt.Printf("\t+ reset machine-controller managed machines %q\n", machine.Name)
		}
	} else {
		s.Logger.Warnln("Failed to list Machines. Worker nodes will not be deleted. If there are worker nodes in the cluster, you might have to delete them manually.")
	}
```

Reply:
fixed in: https://github.com/kubermatic/kubeone/pull/1251/commits/90cccce01661e7def9b492ba2361bd416baba4d3